src/onefetch/image_backends/iterm2.rs

```diff
@@ -0,0 +1,61 @@
+use {
+    image::{imageops::FilterType, DynamicImage, GenericImageView},
+    libc::{
+        ioctl, winsize, STDOUT_FILENO, TIOCGWINSZ,
+    },
+    std::env,
+};
+
+pub struct ITerm2Backend {}
+
+impl ITerm2Backend {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn supported() -> bool {
+        let term_program = env::var("TERM_PROGRAM").unwrap_or("".to_string());
+        term_program == "iTerm.app"
```

Comment:
What about other terminals that supports Iterm2 image protocol, like WezTerm ?

Reply:
WezTerm sets TERM_PROGRAM=WezTerm
It supports iTerm2 image protocol, but it behaves a bit differently from iTerm2:
![スクリーンショット 2020-11-02 20 54 30](https://user-images.githubusercontent.com/107537/97865633-db11db80-1d4d-11eb-8b08-c8434ac5b490.png)

Reply:
if you do this
```rust
image_data.extend(format!("\x1B[{}A", image_rows as u32).as_bytes()); // move cursor to top-left corner
image_data.extend(format!("\x1B[{}C", image_columns as u32 + 1).as_bytes()); // move cursor to top-right corner of image
```
instead of 
```rust
image_data.extend(format!("\x1B[{}A", image_rows as u32 - 1).as_bytes()); // move cursor to start of image`
```
you get the correct input:

![iterm](https://user-images.githubusercontent.com/13710835/97873552-172a4980-1d18-11eb-8a88-6846400cd7f2.png)

Hopefully, it doesn't break the output on Iterm2 :worried:

Reply:
> Still, my first remark remains relevant: we shouldn't assume that Iterm2 will be the only terminal supporting this protocol.

I think we cannot detect all terminals supporting iTerm2 image protocol. 
So I think we should not prohibit them in image_backends::check_if_supported().
Fixed by d821bc35

Reply:
> Logically, this should break the output on Iterm2, in that case - @yoichi plz test it -, I'll open an issue on WezTerm's Github.

I've tested with the change

```
diff --git a/src/onefetch/image_backends/iterm2.rs b/src/onefetch/image_backends/iterm2.rs
index 0c2f3c7..86dc2cf 100644
--- a/src/onefetch/image_backends/iterm2.rs
+++ b/src/onefetch/image_backends/iterm2.rs
@@ -33,7 +33,7 @@ impl super::ImageBackend for ITerm2Backend {
             (lines.len() as f64 / height_ratio) as u32,
             FilterType::Lanczos3,
         );
-        let _image_columns = width_ratio * f64::from(image.width());
+        let image_columns = width_ratio * f64::from(image.width());
         let image_rows = height_ratio * f64::from(image.height());
 
         let mut bytes: Vec<u8> = Vec::new();
@@ -47,7 +47,8 @@ impl super::ImageBackend for ITerm2Backend {
         image_data.extend(encoded_image.bytes());
         image_data.extend(b"\x07");
 
-        image_data.extend(format!("\x1B[{}A", image_rows as u32 - 1).as_bytes()); // move cursor to start of image
+        image_data.extend(format!("\x1B[{}A", image_rows as u32).as_bytes()); // move cursor to top-left corner
+        image_data.extend(format!("\x1B[{}C", image_columns as u32 + 1).as_bytes()); // move cursor to top-right corner of image
         let mut i = 0;
         for line in &lines {
             image_data.extend(format!("\x1B[s{}\x1B[u\x1B[1B", line).as_bytes());
```

fixes the problem on WezTerm, and cause a problem on iTerm2:

![スクリーンショット 2020-11-03 0 43 45](https://user-images.githubusercontent.com/107537/97887763-ae6dbc00-1d6d-11eb-9af8-72bc015bb474.png)

Reply:
Thanks @yoichi, I created the issue --> https://github.com/wez/wezterm/issues/317